src/Common/Config/ConfigProcessor.cpp

```diff
@@ -410,6 +411,47 @@ void ConfigProcessor::doIncludesRecursive(
     }
 }
 
+std::string printXMLIndent(int layer) {
+    std::string indent = "";
+    for (int i = 0; i < layer; ++i) {
+        indent += "  ";
+    }
+    return indent;
+}
+
+void processYAML(YAML::Node node, std::ofstream &fout, std::string path, int layer = 0) {
```

Comment:
I think better not to convert files `.yaml` -> `.xml`, but write something like the class `Poco::XML::DOMParser`, which will read `.yaml` and returns `Poco::XML::Document`:

```
class YAMLParser
{
  Poco::XML::Document* parse(const String& path);
};
```

This class will allow to make other changes very simple, for example, all you will have to change in `ConfigProcessor::processConfig()` is to add three lines:

```
XMLDocumentPtr ConfigProcessor::processConfig(...)
{
    ...
    if (fs::exists(path))
    {
        if (fs::path(path).extension() == ".yaml")
          config = yaml_parser.parse(path);
        else
          config = dom_parser.parse(path);
    }
    ...
}
```

Reply:
Preprocessed files can be always written in XML.

Reply:
OK, then I'll write a new class in a file separate from ConfigProcessor. There is one question though. Is it necessary to add a new class to the namespace DB or not?